src/sentry/integrations/github/client.py

```diff
@@ -224,6 +224,25 @@ def get_commit(self, repo: str, sha: str) -> Any:
         """
         return self.get_cached(f"/repos/{repo}/commits/{sha}")
 
+    def get_merge_commit_sha_from_commit(self, repo: str, sha: str) -> str | None:
+        """
+        Get the merge commit sha from a commit sha.
+        """
+        response = self.get_pullrequest_from_commit(repo, sha)
+        if not response or (isinstance(response, list) and len(response) != 1):
+            # the response should return a single merged PR, return if multiple
+            return None
```

Comment:
What does it mean if we return on multiple? Should we be logging it?

Reply:
for GitHub, this means there are multiple open PRs with the commit https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit
<img width="527" alt="Screenshot 2024-07-19 at 09 53 28" src="https://github.com/user-attachments/assets/7fdd1c44-0abd-4b68-9538-d9902c7eee99">